pkg/client/ipv6.go

```diff
@@ -0,0 +1,19 @@
+package client
+
+import "net"
+
+var hasIPv6Support bool
+
+// HasIPv6Support tells if the system supports IPv6 or not.
+// This doesn't guarantee connections will be successful.
+func HasIPv6Support() bool {
+	return hasIPv6Support
+}
+
+func init() {
+	l, err := net.Listen("tcp6", "::1")
```

Comment:
This confuses me. Atm I do not think that we need this as all it does (as I read it) is to confirm that the system has IPv6 support, why do we care about that? At this stage I would think that a https connection is more relevant, something like:
```go
dialer := &net.Dialer{
        Timeout:   5 * time.Second,
        KeepAlive: 5 * time.Second,
    }

    http.DefaultTransport.(*http.Transport).DialContext = func(ctx context.Context, network, addr string) (net.Conn, error) {
        if addr == "google.com:443" {
            addr = "[2a00:1450:400d:807::200e]:443"
        }
        return dialer.DialContext(ctx, network, addr)
    }
    resp, err := http.Head("https://google.com")
if err!=nil{
    hasIPv6Support = false
}
hasIPv6Support = true
```
well, something along those lines, with checks and fixes.... . As I see it in order to deal with IPv6 only NSs (again I am yet to see such an entity) we need IPV6 connectivity, not only support from kernel.

Reply:
It prevents fatal failures on github's CI which doesn't support IPv6 at all. for non-functional ipv6 I have a client middleware that filters out AAAA records, to be injected by the DNS resolver based on config

Reply:
@karasz

no test reliably can tell us if all IPv6 connections will succeed. here we test for kernel support which will tell us if they will 100% fail, which allows us to run unit tests on github. `client.HasIPv6Support()` has a comment about this on its description.

I believe a fair approach is to allow resolvers to disable AAAA by configuration, otherwise assume that having IPv6 implies it's usable. 

To disable `AAAA` the `RootLookuper` can be assembled using `client.NoAAAA` (#60) as `client.Client`.